server/api/reviews.go

```diff
@@ -0,0 +1,122 @@
+package handler
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/codeday-labs/bookworms/server/db"
+	"github.com/codeday-labs/bookworms/server/utils"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type ReviewBody struct {
+	UserName   string   `json:"user_name"`
+	BookName   string   `json:"book_name"`
+	Text       string   `json:"text"`
+	Categories []string `json:"categories"`
+}
+
+// singular collection name from mongodb collections conventions
+const ReviewsCollection = "review"
+
+func filterReviews(filter interface{}) ([]*db.Review, error) {
+
+	// a slice to store decoded reviews
+	var reviews []*db.Review
+
+	cur, err := db.DB().Collection(ReviewsCollection).Find(db.Ctx, filter)
+
+	// close db connection
+	defer db.DB().Client().Disconnect(db.Ctx)
+
+	// once once done iterating the cursor close
+	defer cur.Close(db.Ctx)
+
+	if err != nil {
+		return reviews, err
+	}
+
+	// Iterate over a returned cursor and decode each at a time
+
+	for cur.Next(db.Ctx) {
+		var r db.Review
+		err := cur.Decode(&r)
+		if err != nil {
+			return reviews, err
+		}
+
+		reviews = append(reviews, &r)
+	}
+
+	if err := cur.Err(); err != nil {
+		return reviews, err
+	}
+
+	if len(reviews) == 0 {
+		return reviews, mongo.ErrNoDocuments
```

Comment:
I would recommend not to return this error as no documents is not an error in our use case.